dbms/src/Flash/tests/gtest_compute_server.cpp

```diff
@@ -286,5 +291,50 @@ try
 }
 CATCH
 
+TEST_F(ComputeServerRunner, cancelAggTasks)
+try
+{
+    startServers(4);
+    {
+        auto [start_ts, res] = injectCancel(context
+                                                .scan("test_db", "test_table_1")
+                                                .aggregation({Max(col("s1"))}, {col("s2"), col("s3")})
+                                                .project({"max(s1)"}));
+        MockComputeServerManager::instance().cancelQuery(start_ts);
+        assertQueryCancelled(start_ts);
```

Comment:
How to ensure that query is cancelled instead of finished.

Reply:
This is the tricky part, the lifetime of a query is related to the BlockInputStreams returned by executeMPPQueryWithMultipleContext.

Reply:
Maybe implement a special TableScan that keeps producing data.